mne_bids_pipeline/steps/preprocessing/_08b_apply_ssp.py

```diff
@@ -81,13 +71,85 @@ def apply_ssp(
     return _prep_out_files(exec_params=exec_params, out_files=out_files)
 
 
+def get_input_fnames_apply_ssp_raw(
+    *,
+    cfg: SimpleNamespace,
+    subject: str,
+    session: Optional[str],
+    run: str,
+    task: Optional[str],
+) -> dict:
+    in_files = _get_run_rest_noise_path(
+        cfg=cfg,
+        subject=subject,
+        session=session,
+        run=run,
+        task=task,
+        kind="filt",
+        mf_reference_run=cfg.mf_reference_run,
+    )
+    assert len(in_files)
+    in_files["proj"] = _proj_path(cfg=cfg, subject=subject, session=session)
+    return in_files
+
+
+@failsafe_run(
+    get_input_fnames=get_input_fnames_apply_ssp_raw,
+)
+def apply_ssp_raw(
+    *,
+    cfg: SimpleNamespace,
+    exec_params: SimpleNamespace,
+    subject: str,
+    session: Optional[str],
+    run: str,
+    task: Optional[str],
+    in_files: dict,
+) -> dict:
+    projs = mne.read_proj(in_files.pop("proj"))
+    in_key = list(in_files.keys())[0]
+    assert in_key.startswith("raw"), in_key
+    raw_fname = in_files.pop(in_key)
+    assert len(in_files) == 0, in_files.keys()
+    raw = mne.io.read_raw_fif(raw_fname)
+    raw.add_proj(projs)
+    out_files = dict()
+    out_files[in_key] = raw_fname.copy().update(processing="clean")
+    msg = f"Writing {out_files[in_key].basename} …"
+    logger.info(**gen_log_kwargs(message=msg))
+    raw.save(out_files[in_key], overwrite=True, split_size=cfg._raw_split_size)
+    _update_for_splits(out_files, in_key)
+    # Report
+    with _open_report(
+        cfg=cfg,
+        exec_params=exec_params,
+        subject=subject,
+        session=session,
+        run=run,
+        task=task,
+    ) as report:
+        msg = "Adding cleaned raw data to report"
+        logger.info(**gen_log_kwargs(message=msg))
+        _add_raw(
+            cfg=cfg,
+            report=report,
+            bids_path_in=out_files[in_key],
+            title="Raw (clean)",
+            tags=("clean",),
+            raw=raw,
+        )
+    return _prep_out_files(exec_params=exec_params, out_files=out_files)
+
+
 def get_config(
     *,
     config: SimpleNamespace,
+    subject: str,
 ) -> SimpleNamespace:
     cfg = SimpleNamespace(
+        processing="filt" if config.regress_artifact is None else "regress",
```

Comment:
this is the third time I've seen the same inline "filt-or-regress" if-else. Makes me worry about forgetting to copy it where it's needed, when adding new code. Not sure if there is a sensible way to reduce that cognitive overhead / duplication that wouldn't feel over-engineered, though.